### PR TITLE
Setup code style fix action

### PR DESCRIPTION
**Pull request by fisker**
https://github.com/prettier/prettier/pull/9051#issuecomment-680122099

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: .github/workflows/code-style.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Fix Code Style
+        run: yarn fix
+
+      - name: Commit Changes
+        uses: stefanzweifel/git-auto-commit-action@v2.3.0
+        with:
+          commit_message: Apply Prettier changes
+          branch: ${{ github.head_ref }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (evilebottnawi) [inline .github/workflows/code-style.yml:31]: Hope it will not create broken code due some bugs on our side or eslint :smile:

Comment (fisker) [reply .github/workflows/code-style.yml:31]: Only run on pull request, don't worry.

**Comment by thorn0** [comment]
This needs some kind of protection against Prettier's idempotence bugs, otherwise it might end up in an infinite loop.

**Comment by fisker** [comment]
It seems a dead end, can't commit to the fork.

File: src/language-css/printer-postcss.js [hunk 2]
```diff
@@ -345,7 +345,8 @@ function genericPrint(path, options, print) {
     }
     case "media-feature": {
       return maybeToLowerCase(
-        adjustStrings(node.value.replace(/ +/g, " "), options)
+        adjustStrings(
+        	node.value.replace(/[ ]{1,}/g, " "), options)
```
Comment (evilebottnawi) [inline src/language-css/printer-postcss.js:349]: Why it changed?

**Comment by fisker** [comment]
Have you seen this https://github.com/stefanzweifel/git-auto-commit-action#using-the-action-in-forks-from-public-repositories ?

**Comment by thorn0** [comment]
Yep, but as you can see we came very close: `remote: Permission to fisker/prettier.git denied to github-actions[bot].`

> You can only make commits on pull request branches that:
> - are opened in a repository that **you have push access** to and that were created from a fork of that repository
> - are on a user-owned fork
> - have permission granted from the pull request creator
> - don't have branch restrictions that will prevent you from committing

([source](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork))

At this point, we can probably give push access to `github-actions[bot]` or one of us can generate a personal access token and add it to the repo as a secret, but both of these options look extremely insecure to me.

**Comment by fisker** [comment]
I agree.

**Comment by thorn0** [comment]
In this PR, you added a new workflow file and it got immediately executed. Is it because you have push access? Or does it work the same way for anyone's PRs? In the latter case, anybody can steal all the secrets by simply opening a PR.

**Comment by fisker** [comment]
It's not working, that commit caused by opening a new PR on my fork 😄 

https://github.com/fisker/prettier/pull/711

**Comment by fisker** [comment]
> In this PR, you added a new workflow file and it got immediately executed. 

I don't know.

**Comment by thorn0** [comment]
It didn't work because with `pull_request_target`, [the workflow files are taken from the base branch](https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks) of the PR (i.e. `master` in our case). We don't have this workflow in `master` yet, that's why it doesn't work.

**Comment by thorn0** [comment]
I guess we should switch our existing workflows from `pull_request` to `pull_request_target`.

**Comment by thorn0** [comment]
Seems to work! Let me open one more PR against the `tmp-9057` branch to test it one more time.

**Comment by fisker** [comment]
Cool!

**Review by lipis — APPROVED** [review]

**Comment by lipis** [comment]
Interesting one..! Are we waiting anything for this one? Worst case scenario we can always remove it later.

**Comment by fisker** [comment]
https://github.com/prettier/prettier/pull/9083

**Comment by fisker** [comment]
Close in favor of #15749

**Pull request closed without merging**